src/test/shared/sam/debugger/samDebugConfigProvider.test.ts

```diff
@@ -246,30 +282,76 @@ describe('AwsSamDebugConfigurationProvider', async () => {
                     //projectRoot: 'root as in beer'
                 },
             })
-            assert.strictEqual(resolved!.name, name)
+            // TODO: why not respect caller-chosen name?
+            assert.strictEqual(resolved!.name, 'SamLocalDebug')
         })
-        it('target=code', async () => {
-            const debugConfig = {
+
+        it('target=code: javascript', async () => {
+            const appDir = path.join(getProjectDir(), 'integrationTest-samples/js-manifest-in-root/')
+            const folder = getWorkspaceFolder(appDir)
+            const c = {
                 type: AWS_SAM_DEBUG_TYPE,
                 name: 'whats in a name',
                 request: DIRECT_INVOKE_TYPE,
                 invokeTarget: {
                     target: CODE_TARGET_TYPE,
-                    lambdaHandler: 'sick handles',
-                    projectRoot: 'root as in beer',
+                    lambdaHandler: 'my.test.handler',
+                    projectRoot: 'src',
                 },
                 lambda: {
                     runtime: validRuntime,
                 },
             }
-            assert.deepStrictEqual(
-                await debugConfigProvider.resolveDebugConfiguration(undefined, debugConfig),
-                debugConfig
-            )
+            ;(c as any).configOnly = true
+            const actual = (await debugConfigProvider.resolveDebugConfiguration(folder, c))!!
+            const expected: SamLaunchRequestArgs = {
```

Comment:
Tests the full CONFIG phase encapsulated in `resolveDebugConfiguration()`. 

- Fast
- Maximum test coverage without actually executing `sam`

Reply:
Why are the emitted payloads so large? Seems a much narrower amount of data is needed in order to instruct VS Code to perform the debug session.

Reply:
there's a lot of opportunity to reduce the payloads, but that should wait until later.